processing/src/main/java/org/apache/druid/query/aggregation/SerializablePairLongDoubleSerde.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.aggregation;
+
+import org.apache.druid.collections.SerializablePair;
+import org.apache.druid.common.config.NullHandling;
+import org.apache.druid.segment.data.ObjectStrategy;
+
+import javax.annotation.Nullable;
+import java.nio.ByteBuffer;
+import java.util.Comparator;
+
+/**
+ * The class serializes a Long-Double pair (SerializablePair<Long, Double>).
+ * The serialization structure is: Long:Double
+ * <p>
+ * The class is used on first/last Double aggregators to store the time and the first/last Double.
+ * (Long:timestamp, Float:value)
+ */
+public class SerializablePairLongDoubleSerde extends AbstractSerializableLongObjectPairSerde<SerializablePairLongDouble>
+{
+  public static final String TYPE_NAME = "serializablePairLongDouble";
+
+  /**
+   * Since SerializablePairLongDouble is subclass of SerializablePair<Long,Double>,
+   * it's safe to declare the generic type of comparator as SerializablePair<Long,Double>.
+   */
+  private static final Comparator<SerializablePair<Long, Double>> VALUE_COMPARATOR = SerializablePair.createNullHandlingComparator(
+      Double::compare,
+      true
+  );
+
+  public SerializablePairLongDoubleSerde()
+  {
+    super(SerializablePairLongDouble.class);
+  }
+
+  @Override
+  public String getTypeName()
+  {
+    return TYPE_NAME;
+  }
+
+  @Override
+  public ObjectStrategy getObjectStrategy()
+  {
+    return new ObjectStrategy<SerializablePairLongDouble>()
```

Comment:
We can use a static ObjectStrategy since it is stateless. It seems right now we are creating a new object for every deserialization.

Reply:
done.